src/profile/components/ChangePasswordForm.vue

```diff
@@ -63,28 +60,45 @@
 <script>
 import validators from "@/shared/form-validators";
 export default {
-  props: ["errorMessage"],
-  data: () => ({
-    password: {
+  props: {
+    isButtonDisabled: Boolean,
+  },
+  data() {
+    return {
       currentPassword: "",
       newPassword: "",
-    },
-    formValidity: false,
-    confirmPassword: null,
-    showCurrentPassword: false,
-    showNewPassword: false,
-    showRepeatPassword: false,
-    passwordRules: [
-      validators.required("Password is required"),
-      validators.min("Min 6 characters", 6),
-    ],
-  }),
+      formValidity: false,
+      confirmPassword: null,
+      showCurrentPassword: false,
+      showNewPassword: false,
+      showRepeatPassword: false,
+      passwordRules: [
+        validators.required("Password is required"),
+        validators.min("Min 6 characters", 6),
+      ],
+      newPasswordRules: [
+        validators.required("Password is required"),
+        validators.min("Min 6 characters", 6),
+        validators.mustNotMatch(
+          "Password must not match!",
```

Comment:
That good but I have confused, I don't know the password should not match which one?
```suggestion
          "New password and existing password should not be the same!",
```